types/rlp_unmarshal_storage.go

```diff
@@ -67,8 +96,17 @@ func (t *Transaction) UnmarshalStoreRLPFrom(p *fastrlp.Parser, v *fastrlp.Value)
 		return err
 	}
 
-	if len(elems) < 2 {
-		return fmt.Errorf("incorrect number of elements to decode transaction, expected 2 but found %d", len(elems))
+	// come TransactionType first if exist
+	if len(elems) != 2 && len(elems) != 3 {
+		return errors.New("expected 2 or 3 elements")
```

Comment:
I find this error more precise but would be helpful to maintain the elems count.

Reply:
Done